### PR TITLE
fix: issue #582 capabilities and dependencies to v9

### PR DESCRIPTION
**Pull request by Sven-Seyfert**
Hi folks 👋 ,

this should hopefully fix the issue #582 , regarding capabilities and dependencies updates to v9 of WDIO.

```
Could not resolve dependency:
peer @wdio/cucumber-framework@"^8.0.0" from wdio-cucumberjs-json-reporter@5.2.1
node_modules/wdio-cucumberjs-json-reporter
wdio-cucumberjs-json-reporter@"5.2.1" from the root project
```

Please notice: I could not test it properly, because I do not know how exactly. I really just applied (hope so) the [hint/suggestion](https://github.com/webdriverio-community/wdio-cucumberjs-json-reporter/pull/583#issuecomment-2305334613) of @christian-bromann . Maybe I have to change or add commits here, to fulfill the aspects of doing the pull requests well. If so, please do not hesitate to say it. I am grateful to learn and I am grateful of the chance to help/contribute.

Best regards
Sven

### CODE REVIEW

**Comment by christian-bromann** [comment]
Thanks for taking a stab at this. There seem to be some unit tests failing. Let me know if you have any questions.

**Comment by Sven-Seyfert** [comment]
Hi @christian-bromann 👋 ,

to be honest, there are obvious missings like to change `WebDriver.DesiredCapabilities` to `WebdriverIO.Capabilities` in `metadata.test.ts` too. Which I have done now (see new commit). But there are also things like the snapshots that I don't know about at all 😕 . Do I have to recreate the snapshots? I guess the unit tests should be almost the same, otherwise I would adjust them for the sake of being green/pass, but that is not the intention behind it, for sure.

In other words, I do not understand some concepts that are crusial I believe. Either we would have a broader discussion and probably another helping hand by you or some other person, or I have to close the PR because **I don't want to break things** that could be overseen in the review process.

_#lack-of-skill_

==> I am sorry. Maybe you or another person can continue on this PR?

Best regards
Sven

**Comment by Sven-Seyfert** [comment]
==> I guess this PR can be closed, because Hanna already introduced another one #608 😄.

**Comment by Sven-Seyfert** [comment]
I will close this PR because Hannas #608 is far more promising. Also I try to support their too.
